src/components/CompareResults/ResultsView.tsx

```diff
@@ -34,6 +45,21 @@ function ResultsView(props: ResultsViewProps) {
     document.title = title;
   }, [title]);
 
+  useEffect(() => {
+    const waitForData = async () => {
+      await results;
+      setLoading(false);
+    };
+
+    void waitForData();
+  }, [results]);
+
+  const handleLoaderRefresh = () => {
+    setLoading(true);
+    const currentUrl = `/compare-results/${location.search}`;
+    fetcher.load(currentUrl);
```

Comment:
Are you sure the explicit load is useful? My understanding is that react-router should handle that directly with its `Form` component already.

(same for the other view)

If we get rid of this + the loading boolean, we can also remove all of the changes related to handleRefresh.

Reply:
So I looked up the `Form` API and it allows us to reload the document. I have this reload only for the edit feature. I think this solution is the best for the the edit feature. What do you think?

```
   <Form
          action='/compare-over-time-results'
          onSubmit={onFormSubmit}
          className='form-wrapper'
          aria-label='Compare over time form'
          reloadDocument={hasEditButton ?? true}
        >
//// rest of code
```

Reply:
Good find!

However I'm not so sure, because the full page is redrawing, as a user I find this quite distracting. I much prefered the previous UX.